core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -804,18 +804,17 @@ private DeleteResult innerDelete(Delete delete) throws IOException {
 
             final long expectedVersion = delete.version();
 
-            final Optional<DeleteResult> result =
+            final Optional<DeleteResult> checkVersionConflictResult =
```

Comment:
nit: shall we call this resultOnVersionConflict? also - it would be good to do the same in innerIndex